lib/s3/models/BucketConfig.rb

```diff
@@ -172,6 +172,23 @@ def pretty_json
         }.reject { |k, v| v.nil? })
       end
 
+      # Public: deep sort an array of hashes or hash of arrays or array of arrays or hash of hashes or... you get the idea
+      #
+      #
+      # obj - The array or hash to deep sort
+      #
+      # Returns a deep sorted array of hashes or hash of arrays
+      def deep_sort_hash_array(obj)
+        case obj
+        when Array
+          obj.map!{|v| deep_sort_hash_array(v)}.sort_by!{|v| (v.to_s rescue nil) }
```

Comment:
this should probably just be 'map' and 'sort_by' instead of 'map!' and 'sort_by!'